docs/standard/base-types/regular-expression-example-scanning-for-hrefs.md

```diff
@@ -14,46 +14,48 @@ helpviewer_keywords:
-  
+## The Regex object
+
+Because the `DumpHRefs` method can be called multiple times from user code, it uses the `static` (`Shared` in Visual Basic) <xref:System.Text.RegularExpressions.Regex.Match%28System.String%2CSystem.String%2CSystem.Text.RegularExpressions.RegexOptions%29?displayProperty=nameWithType> method. This enables the regular expression engine to cache the regular expression and avoids the overhead of instantiating a new <xref:System.Text.RegularExpressions.Regex> object each time the method is called. A <xref:System.Text.RegularExpressions.Match> object is then used to iterate through all matches in the string.
+
+:::code language="csharp" source="snippets/regular-expression-example-scanning-for-hrefs/csharp/Program.cs" id="regex":::
+:::code language="vb" source="snippets/regular-expression-example-scanning-for-hrefs/vb/Program.vb" id="regex":::
+
+The following example then illustrates a call to the `DumpHRefs` method.
+
+:::code language="csharp" source="snippets/regular-expression-example-scanning-for-hrefs/csharp/Program.cs" id="main":::
+:::code language="vb" source="snippets/regular-expression-example-scanning-for-hrefs/vb/Program.vb" id="main":::
+
+The regular expression pattern `href\s*=\s*(?:["'](?<1>[^"']*)["']|(?<1>[^>\s]+))` is interpreted as shown in the following table.
+
+| Pattern                | Description                                                     |
+|------------------------|-----------------------------------------------------------------|
+| `href`                 | Match the literal string "href". The match is case-insensitive. |
+| `\s*`                  | Match zero or more white-space characters.                      |
+| `=`                    | Match the equals sign.                                          |
+| `\s*`                  | Match zero or more white-space characters.                      |
+| `(?:`                  | Start a non-capturing group.                                    |
+| `["'](?<1>[^"']*)["']` | Match a quotation mark or apostrophe, followed by a capturing group that matches any character other than a quotation mark or apostrophe, followed by a quotation mark or apostrophe. The group named `1` is included in this pattern. |
+| `|`                    | Boolean OR that matches either the previous expression or the next expression. |
+| `(?<1>[^>\s]+)`        | A capturing group that uses a negated set to match any character other than a greater-than sign or a whitespace character. The group named `1` is included in this pattern.|
+| `)`                    | End the non-capturing group.                                    |
```

Comment:
@luisquintanilla this is the only section really changed. Everything else was just formatting updates like removing doublespace at the end of lines and removing bad indentation.